homeassistant/components/rainmachine/__init__.py

```diff
@@ -152,25 +154,26 @@
     websession = aiohttp_client.async_get_clientsession(hass)
 
     try:
-        client = await login(
-            config_entry.data[CONF_IP_ADDRESS],
-            config_entry.data[CONF_PASSWORD],
-            websession,
-            port=config_entry.data[CONF_PORT],
-            ssl=config_entry.data[CONF_SSL])
-        rainmachine = RainMachine(
-            client,
-            config_entry.data.get(CONF_BINARY_SENSORS, {}).get(
-                CONF_MONITORED_CONDITIONS, list(BINARY_SENSORS)),
-            config_entry.data.get(CONF_SENSORS, {}).get(
-                CONF_MONITORED_CONDITIONS, list(SENSORS)),
-            config_entry.data.get(CONF_ZONE_RUN_TIME, DEFAULT_ZONE_RUN)
-        )
-        await rainmachine.async_update()
-    except RainMachineError as err:
+        with async_timeout.timeout(4):
```

Comment:
For requests timeout we usually want the library that does the requests to implement the timeout. What do you think?

Reply:
I was curious about that. I saw other platforms do it directly (Hue comes to mind), but if you don’t like that, I can do it in the library).

Reply:
I would suggest adding a parameter to the login function for timeout.

Reply:
@MartinHjelmare: Via cbbae84, I've moved all the timeout logic into `regenmaschine`.